src/drivers/imx/sdma.c

```diff
@@ -486,16 +486,10 @@ static int sdma_start(struct dma_chan_data *channel)
 
 	tracev_sdma("sdma_start(%d)", channel->index);
 
-	if (channel->index) {
-		/* rest of the channels are prepared first */
-		if (channel->status != COMP_STATE_PREPARE &&
-		    channel->status != COMP_STATE_SUSPEND)
-			return -EINVAL;
-	} else {
-		/* channel 0 doesn't get set_config set on */
-		if (channel->status != COMP_STATE_READY)
-			return -EINVAL;
-	}
+	/* rest of the channels are prepared first */
```

Comment:
This one comment may be unnecessary as it will lack context after the change.

Reply:
yes, will remove.